e2e/snapshot_e2e/k8s.py

```
@@ -167,6 +179,23 @@ def exec_command(
+
+def exec_payload(namespace: str, pod: str, command: str) -> str:
+    """Exec output with whatever the login shell printed first dropped.
+
+    exec_command merges stderr into the stream, so a container whose profile
+    writes anything breaks every caller that parses the result rather than
+    matching a substring in it.
+    """
+    output = exec_command(namespace, pod, f"echo {PAYLOAD_MARKER}; {command}")
+    _, marker, payload = output.partition(PAYLOAD_MARKER)
+    if not marker:
+        raise AssertionError(f"exec output carried no payload marker: {output!r}")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)